apps/server/src/modules/provisioning/strategy/sanis/service/sanis-school.service.ts

```diff
@@ -1,27 +1,25 @@
 import { Injectable } from '@nestjs/common';
 import { SanisResponse } from '@src/modules/provisioning/strategy/sanis/sanis.response';
-import { EntityId, School } from '@shared/domain';
-import { SchoolDto } from '@src/modules/school/uc/dto/school.dto';
+import { EntityId } from '@shared/domain';
 import { ProvisioningSchoolOutputDto } from '@src/modules/provisioning/dto/provisioning-school-output.dto';
 import { SanisResponseMapper } from '@src/modules/provisioning/strategy/sanis/sanis-response.mapper';
-import { SchoolUc } from '@src/modules/school/uc/school.uc';
 import { SchoolRepo } from '@shared/repo';
+import { SchoolDO } from '@shared/domain/domainobject/school.do';
+import { SchoolService } from '@src/modules/school';
 
 @Injectable()
 export class SanisSchoolService {
 	constructor(
 		private readonly responseMapper: SanisResponseMapper,
-		private readonly schoolUc: SchoolUc,
+		private readonly schoolService: SchoolService,
 		private readonly schoolRepo: SchoolRepo
 	) {}
 
-	async provisionSchool(data: SanisResponse, systemId: EntityId): Promise<SchoolDto> {
+	async provisionSchool(data: SanisResponse, systemId: EntityId): Promise<SchoolDO> {
 		const school: ProvisioningSchoolOutputDto = this.responseMapper.mapToSchoolDto(data, systemId);
-		const schoolEntity: School | null = await this.schoolRepo.findByExternalId(school.externalId, systemId);
-		if (schoolEntity instanceof School) {
-			school.id = schoolEntity.id;
-		}
-		const savedSchool: SchoolDto = await this.schoolUc.saveProvisioningSchoolOutputDto(school);
+		const schoolDo: SchoolDO | null = await this.schoolRepo.findByExternalId(school.externalId, systemId);
+		school.id = schoolDo ? schoolDo.id : undefined;
+		const savedSchool: SchoolDO = await this.schoolService.saveProvisioningSchoolOutputDto(school);
```

Comment:
It does have that. We have this Provisioning specific method from an earlier architecture implementation. This will be changed once we refactor provisioning.